noir/noir-repo/compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -714,7 +714,7 @@ pub(crate) fn error_selector_from_type(typ: &ErrorType) -> ErrorSelector {
             typ.hash(&mut hasher);
             let hash = hasher.finish();
             assert!(hash != 0, "ICE: Error type {} collides with the string error type", typ);
-            ErrorSelector::new(hash)
+            ErrorSelector::new(hash as u32) // NOTE: will this break? - i want this to be lossy
```

Comment:
Error selector can be u64, we just need to change in the codegen for the revert that it shouldn't use a usize constant but rather a u64 one